### ISSUE TITLE
Switching between devices?

### ISSUE THREAD

**Issue by TheChilliPL** [Lattyware/massivedecks#186]
Normally, if you get disconnected or something, you are allowed to reconnect, preserving your points, but there's a problem if you want to switch between devices.
I think there should be a way to switch between them, but I'm not sure about how it could be implemented. The first thing that comes into mind is just a login system—either email and password or via Google O-Auth like Many Decks, but it has some issues—some players just don't want to log in and that's understandable in a party game like this one.
Another way could be to force the player to stay on the old device and showing him a notification (or a toast message) once someone with the same name tries to enter the game (with the option to switch to the other device; maybe with some device switching mode to enable, without which the game would display the usual can't-join-with-the-same-name error).

**Comment by Lattyware** [comment]
Yeah, this would be a nice feature, but full on authentication is a bit heavy and counter to way the game is generally played.

Giving a player the ability to authorize another device as you suggest seems like a reasonable way of doing this. The ability for a privileged player to authorize the user taking over another user who has left would probably be the best way to avoid forcing the user to juggle two devices.

Technically, multi-device already exists in the game—we do it with Chromecasts (the Chromecast is just another client with the same authorization as the primary client). However, that is a little different in that it is spectation-only. I imagine trying to actually play from two places at once would cause errors, but that isn't really a problem: I think we can safely say it is user error at that point.

Implementation-wise, it should be pretty simple: just giving the user a token with the same user id should let them play as the same user from multiple devices, so the bulk of this task would just be constructing the UI and flow to authorize it.
